packages/unminify/src/transformations/__tests__/un-jsx.spec.ts

```diff
@@ -201,11 +199,68 @@ const Foo = () => {
 const Foo = () => {
   const Component = g ? "p" : "div";
   const Component_1 = r ? "a" : "div";
-  return <div><Component_1 key="b">bar</Component_1><Component key={c}>baz</Component></div>;
+  return <div><div key="b">bar</div><div key={c}>baz</div></div>;
 };
 `,
 )
 
+inlineTest('jsx with dynamic Component tag #4',
+  `
+function fn() {
+  const value = "wakaru"
+  return React.createElement(
+    value ? "a" : "div",
```

Comment:
I think I have added the dynamic components test case.

Reply:
This test is ok. But the value and `const Component` should be removed. I know it would be too complex to implement, you can leave it to me 😄

Reply:
haha, I can do it a try to remove this.

Reply:
check `removeDeclarationIfUnused()`